.github/workflows/build.yml

```diff
@@ -23,45 +24,52 @@ jobs:
       with:
         dotnet-version: 6.0.x
 
-    - name: Restore dependencies
-      run: dotnet restore
-
-    - name: Main Build
-      run: dotnet build --configuration $config --no-restore -p:FileVersionRevision=$GITHUB_RUN_NUMBER
-
-    - name: Docker Build - Chat Bot Sample
-      run: docker build -f samples/chat/csharp-inproc/Dockerfile -t openai-func-sample-csharp-inproc .
+    - name: Docker Build - Chat Bot Sample Out of proc
+      run: docker build -f samples/chat/csharp-ooproc/Dockerfile -t openai-func-sample-csharp-ooproc .
 
     - name: Docker Run - Azurite
       run: docker run -d -p 10000:10000 -p 10001:10001 -p 10002:10002 mcr.microsoft.com/azure-storage/azurite
 
-    - name: Docker Run - Chat Bot Sample
+    - name: Docker Build - Chat Bot Sample Out of proc
+      run: docker build -f samples/chat/csharp-ooproc/Dockerfile -t openai-func-sample-csharp-ooproc .
+
+    - name: Docker Run - Chat Bot Sample Out of Proc
       env:
         AZURE_OPENAI_KEY: ${{ secrets.AZURE_OPENAI_KEY }}
         AZURE_OPENAI_ENDPOINT: ${{ vars.AZURE_OPENAI_ENDPOINT }}
         AZURE_DEPLOYMENT_NAME: ${{ vars.AZURE_DEPLOYMENT_NAME }}
       run: |
-          docker run -d -p 7071:80 --name openai-func-sample-csharp-inproc \
+          docker run -d -p 7071:80 --name openai-func-sample-csharp-ooproc \
             --add-host host.docker.internal:host-gateway \
             --env AZURE_OPENAI_KEY=$AZURE_OPENAI_KEY \
             --env AZURE_OPENAI_ENDPOINT=$AZURE_OPENAI_ENDPOINT \
             --env AZURE_DEPLOYMENT_NAME=$AZURE_DEPLOYMENT_NAME \
             --env 'AzureWebJobsStorage=UseDevelopmentStorage=true;DevelopmentStorageProxyUri=http://host.docker.internal' \
-            openai-func-sample-csharp-inproc
+            openai-func-sample-csharp-ooproc
 
     - name: Sleep for 10 seconds
       run: sleep 10s
 
-    - name: E2E Test - Chat Bot Sample
+    - name: Create new directory
+      run: mkdir ~/NuGetPackagesLocal
+
+    - name: Create/Copy NuGet Package to local directory and Build
+      run: |
+        dotnet build /home/runner/work/azure-functions-openai-extension/azure-functions-openai-extension/src/WebJobs.Extensions.OpenAI/WebJobs.Extensions.OpenAI.csproj --configuration $config -p:FileVersionRevision=$GITHUB_RUN_NUMBER -p:WebJobsVersion=$newVersion -p:Version=$newVersion
+        sudo cp "/home/runner/work/azure-functions-openai-extension/azure-functions-openai-extension/src/WebJobs.Extensions.OpenAI/bin/Release/Microsoft.Azure.WebJobs.Extensions.OpenAI.${newVersion}.nupkg" ~/NuGetPackagesLocal
+
+        dotnet nuget add source ~/NuGetPackagesLocal
+        dotnet nuget list source
+
+        dotnet build --configuration $config /p:WebJobsVersion=$newVersion;FileVersionRevision=$GITHUB_RUN_NUMBER
```

Comment:
Given that the Docker build has already succeeded in the previous step, is there a reason for doing a separate build here in the CI process?

Reply:
Yeah so I tried it without the separate build in the CI process, but it fails to pick up any changes within webjobs. For some reason we need both the docker changes and the CI changes. We probably need both since the docker build changes are within the container only and not within the actual build pipeline, so we need to explicitly build and add the nuget source again.

Reply:
I guess it's because the test project, which is built in the CI process, depends on the built extension.